SYCL/ESIMD/api/functional/ctors/ctor_move.cpp

```diff
@@ -0,0 +1,232 @@
+      // because any concurrent access to the same memory location is UB, even
+      // in case we are updating variable to the same value in multiple threads
+      shared_vector<int> flags_storage(n_checks, flags_allocator);
+
+      // run check for each of the reference elements
+      queue.submit([&](sycl::handler &cgh) {
+        const DataT *const ptr_in = input.data();
+        const auto ptr_out = result.data();
+        const auto ptr_flags = flags_storage.data();
+
+        cgh.parallel_for<KernelName>(
+            range, [=](sycl::id<1> id) SYCL_ESIMD_KERNEL {
+              const auto work_item_index = id[0];
+              const DataT *const in = ptr_in + work_item_index;
+              const auto out = ptr_out + work_item_index;
+              const auto was_moved_flag = ptr_flags + work_item_index;
+              *was_moved_flag = run_check(in, out);
+            });
+      });
+      queue.wait_and_throw();
+
+      // Oversafe: verify the proper signature was called for every check
+      was_moved = std::all_of(flags_storage.cbegin(), flags_storage.cend(),
+                              [](int flag) { return flag; });
+    } else {
+      assert((input.size() == NumElems) &&
+             "Unexpected size of the input vector");
+
+      shared_vector<int> flags_storage(1, flags_allocator);
+
+      queue.submit([&](sycl::handler &cgh) {
+        const DataT *const in = input.data();
+        const auto out = result.data();
+        const auto was_moved_flag = flags_storage.data();
+
+        cgh.single_task<KernelName>(
+            [=]() SYCL_ESIMD_KERNEL { *was_moved_flag = run_check(in, out); });
+      });
+      queue.wait_and_throw();
+
+      was_moved = flags_storage[0];
+    }
+
+    if (!was_moved) {
+      passed = false;
+
+      // TODO: Make ITestDescription architecture more flexible
+      std::string log_msg = "Failed for simd<";
+      log_msg += data_type + ", " + std::to_string(NumElems) + ">";
+      log_msg += ", with context: " + TestCaseT::get_description();
+      log_msg += ". A copy constructor instead of a move constructor was used.";
+
+      log::note(log_msg);
+    } else {
+      for (size_t i = 0; i < reference.size(); ++i) {
+        const auto &retrieved = result[i];
+        const auto &expected = reference[i];
+
+        if (!are_bitwise_equal(retrieved, expected)) {
+          passed = false;
+
+          log::fail(ctors::TestDescription<DataT, NumElems, TestCaseT>(
+              i, retrieved, expected, data_type));
+        }
+      }
+    }
+
+    return passed;
+  }
+
+private:
+  // The core check logic.
+  // Uses USM pointers for input data and to store the data from the new simd
+  // instance, so that we could check it later
+  // Returns the flag that should be true only if the move constructor was
+  // actually called, to differentiate with the copy constructor calls
+  static bool run_check(const DataT *const in, DataT *const out) {
```

Comment:
could you use the same names for `input data` and `output data` as they uses in [copy constructor](https://github.com/intel/llvm-test-suite/blob/intel/SYCL/ESIMD/api/functional/ctors/ctor_copy.cpp#L62)

Reply:
From the point of `run_check` it's an input data it works on. There is no need to know what logic is on the upper level.

Moreover, for the case of `simd<T,1>` such input data represents a `single element from the reference data`, not the `reference data` itself, so using `ref_data` instead of `in` here sounds somewhat misleading.

Does it address your concerns?

Reply:
> Does it address your concerns?

yes, thanks for the clarification